site/content/blog/kapp_deploy_oidc_gke.md

```diff
@@ -0,0 +1,118 @@
+3. Create a service account and connect Workload Identity Pool you just created to the service account by assigning Workload Identity User role. For more information, see the [GCP documentation](https://cloud.google.com/iam/docs/workload-identity-federation?_ga=2.114275588.-285296507.1634918453#conditions).
+
+To update workflows for OIDC, you will need to make two changes to your Github Action YAML:
+
+1. Add permissions settings for the token. The job or workflow run requires a permissions setting with `id-token: write`. You won't be able to request the OIDC JWT ID token if the permissions setting for `id-token` is set to `read` or `none`.
+
+```yaml
+permissions:
+  id-token: write
+```
+
+2. Use the google-github-actions/auth action to exchange the OIDC token (JWT) for a cloud access token.
+
+```yaml
+steps:
+- id: 'auth'
+  name: 'Authenticate to Google Cloud'
+  uses: 'google-github-actions/auth@v0.4.0'
+  with:
+    workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+    service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+```
+
+This will use the configured workload_identity_provider and service_account to authenticate future steps. Make sure to replace the value of `workload_identity_provider` with the path to your identity provider in GCP and also replace the value of `service_account` key with the name of your service account in GCP. 
+
+### Example
+
+Here is a sample Github Action which gets triggered when a new tag is created on the repo. It authenticates with GCP, gets the GKE credentials, install carvel tools on the GKE cluster and deploy a simple app using kapp. 
+
+```yaml
+name: OIDC auth GCP
+
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  Get_OIDC_ID_token:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: 'write'
+
+    steps:
+      # actions/checkout MUST come before auth
+      - uses: 'actions/checkout@v3'
+
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v0.4.0'
+        with:
+          workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+          service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+      - id: get-gke-credentials
+        uses: google-github-actions/get-gke-credentials@v0.4.0
+        with:
+          cluster_name: cluster-yash
+          location: us-central1-a
+
+      - id: install-kapp
+        run: |-
+          wget -O- https://carvel.dev/install.sh > install.sh
+          sudo bash install.sh
+          kapp version
+
+      - id: 'deploy-with-kapp'
+        run: |-
+          kapp deploy -a app -f simple-app.yml -y
+          kapp ls
+```
+
+Please refer to this [Github Repo](https://github.com/sethiyash/carvel-kapp-oidc-github) which contains a Github Action and simple-app.yml which we will deploy on GKE using kapp. 
```

Comment:
can we move this into github.com/k14s org?

Reply:
@renuy I think this would apply to the dagger blog as well.

Reply:
Yeah sure. but looks like I don't have the access to create the repo in K14s can you help me with the access?